opensubtitles/opensubtitles.go

```diff
@@ -0,0 +1,267 @@
+// authentication, provide an http.Client that will perform the authentication
+// for you (such as that provided by the golang.org/x/oauth2 library).
+func NewClient(httpClient *http.Client, token string, cred Credentials) (c *Client) {
+	if httpClient == nil {
+		httpClient = &http.Client{}
+	}
+	baseURL, _ := url.Parse(defaultBaseURL)
+	//uploadURL, _ := url.Parse(uploadBaseURL)
+
+	c = &Client{client: httpClient, BaseURL: baseURL, UserAgent: userAgent, Token: token}
+	c.common.client = c
+	c.Authentication = (*AuthenticationService)(&c.common)
+	c.Discover = (*DiscoverService)(&c.common)
+	c.Download = (*DownloadService)(&c.common)
+	c.Find = (*FindService)(&c.common)
+	c.Info = (*InfoService)(&c.common)
+	c.Search = (*SearchService)(&c.common)
+
+	//Check if struct Credential is not empty
+	if (Credentials{}) != cred {
+		c.Credential = cred
+	}
+
+	return
+}
+
+//Connect return a new Client with a working token by making the authentication
+//with the Authentication Login function.
+func (c *Client) Connect() (*Client, error) {
+	if c == nil {
+		return nil, errors.New("no client provided")
+	}
+	if len(c.Token) < 1 {
+		log, resp, _ := c.Authentication.Login(context.Background(), &c.Credential)
+		if (&LoggedIn{}) != log {
+			if len(log.Token) > 0 && resp.StatusCode == http.StatusOK {
+				c.Token = log.Token
+			} else {
+				return nil, errors.New("Wrong Username/Password")
+			}
+
+		}
+	}
+
+	return c, nil
+}
+
+// NewRequest creates an API request. A relative URL can be provided in urlStr,
+// in which case it is resolved relative to the BaseURL of the Client.
+func (c *Client) NewRequest(method, urlStr string, body string) (*http.Request, error) {
+	u, err := c.BaseURL.Parse(urlStr)
+	if err != nil {
+		return nil, err
+	}
+
+	/*var buf io.ReadWriter
+	if body != nil {
+		buf = &bytes.Buffer{}
+		/*enc := json.NewEncoder(buf)
+		enc.SetEscapeHTML(false)
+		err := enc.Encode(body)
+		if err != nil {
+			return nil, err
+		}
+	}*/
+	var b io.Reader
+	if body != "" {
+		b = strings.NewReader(body)
+	}
+	req, err := http.NewRequest(method, u.String(), b)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Add(headerAccept, mediaTypeJSON)
+	req.Header.Set("Content-Type", mediaTypeJSON)
+	if c.Token != "" {
+		req.Header.Set("Authorization", c.Token)
+	}
+	//req.Header.Set("Accept", mediaTypeV3)
+	return req, nil
+}
+
+// CheckResponse checks the API response for errors, and returns them if
+// present. A response is considered an error if it has a status code outside
+// the 200 range or equal to 202 Accepted.
+// API error responses are expected to have either no response
+// body, or a JSON response body that maps to ErrorResponse. Any other
+// response body will be silently ignored.
+func CheckResponse(r *http.Response, b []byte) error {
+	errorResponse := &ErrorResponse{Response: r}
+	if b != nil {
+		json.Unmarshal(b, errorResponse)
+	}
+	if errorResponse.Errors != nil {
+		return errorResponse
+	}
+	return nil
```

Comment:
Codacy found an issue: [extra empty line at the end of a block](https://app.codacy.com/manual/TheForgotten69/go-opensubtitles/pullRequest?prid=6093402)